pyuvdata/uvdata.py

```diff
@@ -2415,6 +2419,25 @@ def _select_preprocess(self, antenna_nums, antenna_names, ant_str, bls,
                     raise ValueError(
                         'Time {t} is not present in the time_array'.format(t=jd))
 
+        if time_range is not None:
+            if np.array(time_range).size != 2:
```

Comment:
I don't know if it is anymore efficient that this, probably honestly it is the same operation, but numpy does offer a `np.size` function which just returns the size instead of us having to cast as an array then take the size. Not sure it matters in the end, but if you are interested. I've attached a rather bad benchmark, they looks to take about the same mean time but `np.size` has a lower variance at least in python 3.8 with numpy 1.17.4

```
In [4]: %timeit np.size([l for l in range(int(1e5))])                                                                                                                                                               
9.49 ms ± 8.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [5]: %timeit np.array([l for l in range(int(1e5))]).size                                                                                                                                                         
9.57 ms ± 34.3 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [6]: print(np.__version__)                                                                            
1.17.4
```